KarhooUISDK/Screens/CheckoutScreen/Checkout/KarhooCheckoutViewModel.swift

```diff
@@ -143,16 +143,27 @@ final class KarhooCheckoutViewModel: ObservableObject {
 
     func getDateScheduledDescription() -> String {
         let date = journeyDetails.scheduledDate ?? Date()
+        let locale = KarhooCountryParser.getSupportedLocale()
+        dateFormatter.set(locale: locale)
+        
+        if let timezone = journeyDetails.originLocationDetails?.timezone() {
+            dateFormatter.set(timeZone: timezone)
+        }
+        
         let dateFormatted = dateFormatter.display(
             date,
             dateStyle: .long,
             timeStyle: .none
         )
+        
         let weekday = {
             guard let weekdayIndex = Calendar.current.dateComponents([.weekday], from: date).weekday else {
                 return ""
             }
-            return (Calendar.current.standaloneWeekdaySymbols[safe: max(weekdayIndex - 1, 0)] ?? "") + ", "
+            
+            var calendar = Calendar.current
+            calendar.locale = locale
```

Comment:
Shouldn't this value be reset once the calculation is done?

Reply:
Why would this be necessary? Don't we want the calendar to always be shown in one of the supported locales?

Reply:
Well, the method potentially modifies class property behavior, which, as additional hidden functionality, is against the KISS principle. 
Maybe just create a new instance date formatter instance just for this method. This way it may be used however it's best for it. 
Or even better: change the default DateFormatter implementation, to use this `getSupportedLocale` instead of the `Locale.current`